titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/v3/rest/JobManagementResource.java

```diff
@@ -126,6 +130,35 @@ public Response setJobDisruptionBudget(@PathParam("jobId") String jobId,
         return Responses.fromVoidMono(jobManagementClient.updateJobDisruptionBudget(request));
     }
 
+    @PUT
+    @ApiOperation("Update attributes of a job")
+    @Path("/jobs/{jobId}/attributes")
+    public Response updateJobAttributes(@PathParam("jobId") String jobId,
+                                        JobAttributesUpdate request) {
+        JobAttributesUpdate sanitizedRequest;
+        if (request.getJobId().isEmpty()) {
+            sanitizedRequest = request.toBuilder().setJobId(jobId).build();
+        } else {
+            if (!jobId.equals(request.getJobId())) {
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            sanitizedRequest = request;
+        }
+        return Responses.fromVoidMono(jobManagementClient.updateJobAttributes(sanitizedRequest));
+    }
+
+    @DELETE
+    @ApiOperation("Delete attributes of a job with the specified key names")
+    @Path("/jobs/{jobId}/attributes")
+    public Response deleteJobAttributes(@PathParam("jobId") String jobId,
+                                        @QueryParam("keys") final List<String> keys) {
```

Comment:
Shouldn't this be `String` value with comma separated key list?

Reply:
I have seen lists represented both as delimited strings as well as multiple query params with the same name. I just chose the JaxRS way (multiple query params), but I don't have a strong preference.

Reply:
In Titus REST API we encode multiple keys (job ids, task ids) as delimited strings.

Reply:
I will change it to be consistent.